resources/providers/awslib/kms/provider_test.go

```diff
@@ -0,0 +1,191 @@
+				"GetKeyRotationStatus": {{{mock.Anything, mock.Anything}, {nil, errors.New("some error")}}},
+			},
+			expected:    []awslib.AwsResource(nil),
+			expectError: false,
+			regions:     []string{awslib.DefaultRegion},
+		},
+		{
+			name: "Should return a resource for a symmetric key",
+			kmsClientMockReturnVals: kmsClientMockReturnVals{
+				"ListKeys": {
+					{{mock.Anything, mock.Anything}, {&kmsClient.ListKeysOutput{Keys: []types.KeyListEntry{
+						{KeyId: &keyId1},
+					}}, nil}},
+					{{mock.Anything, mock.Anything}, {&kmsClient.ListKeysOutput{Keys: []types.KeyListEntry{
+						{KeyId: &keyId2},
+					}}, nil}},
+				},
+				"DescribeKey": {
+					{{mock.Anything, mock.MatchedBy(MatchDescribeKeyInput(keyId1))}, {&kmsClient.DescribeKeyOutput{KeyMetadata: &types.KeyMetadata{KeyId: &keyId1, KeySpec: types.KeySpecSymmetricDefault}}, nil}},
+					{{mock.Anything, mock.MatchedBy(MatchDescribeKeyInput(keyId2))}, {&kmsClient.DescribeKeyOutput{KeyMetadata: &types.KeyMetadata{KeyId: &keyId2, KeySpec: types.KeySpecSymmetricDefault}}, nil}},
+				},
+				"GetKeyRotationStatus": {
+					{{mock.Anything, mock.MatchedBy(MatchGetKeyRotationStatusInput(keyId1))}, {&kmsClient.GetKeyRotationStatusOutput{KeyRotationEnabled: true}, nil}},
+					{{mock.Anything, mock.MatchedBy(MatchGetKeyRotationStatusInput(keyId2))}, {&kmsClient.GetKeyRotationStatusOutput{KeyRotationEnabled: true}, nil}},
+				},
+			},
+			expected: []awslib.AwsResource{
+				KmsInfo{KeyMetadata: types.KeyMetadata{KeyId: &keyId1, KeySpec: types.KeySpecSymmetricDefault}, KeyRotationEnabled: true},
+				KmsInfo{KeyMetadata: types.KeyMetadata{KeyId: &keyId2, KeySpec: types.KeySpecSymmetricDefault}, KeyRotationEnabled: true},
+			},
+			expectError: false,
+			regions:     []string{"us-east-1", "us-east-2"},
+		},
+	}
+
+	for _, test := range tests {
+		kmsClientMock := &MockClient{}
+		for funcName, returnVals := range test.kmsClientMockReturnVals {
+			for _, vals := range returnVals {
+				kmsClientMock.On(funcName, vals[0]...).Return(vals[1]...).Once()
+			}
+		}
+
+		kmsProvider := Provider{
+			log:     s.log,
+			clients: testhelper.CreateMockClients[Client](kmsClientMock, test.regions),
+		}
+
+		ctx := context.Background()
+
+		results, err := kmsProvider.DescribeSymmetricKeys(ctx)
+		if test.expectError {
+			s.Error(err)
+		} else {
+			s.NoError(err)
+		}
+
+		// Using `ElementsMatch` instead of the usual `Equals` since iterating over the regions map does not produce a
+		//	guaranteed order
+		s.ElementsMatch(test.expected, results, fmt.Sprintf("Test '%s' failed, elements do not match", test.name))
+	}
+}
+
+func MatchDescribeKeyInput(keyId string) func(k *kmsClient.DescribeKeyInput) bool {
+	return func(k *kmsClient.DescribeKeyInput) bool {
+		return *k.KeyId == keyId
+	}
+}
+
+func MatchGetKeyRotationStatusInput(keyId string) func(k *kmsClient.GetKeyRotationStatusInput) bool {
+	return func(k *kmsClient.GetKeyRotationStatusInput) bool {
+		return *k.KeyId == keyId
+	}
+}
```

Comment:
these are used to ensure `DescribeKey` and `GetKeyRotationStatus` are both called with the right `KeyId` (using `mock.MatchedBy`) 

both do the same thing - check `KeyId` equals some string. it'd be nicer to make this a generic function